core/src/main/java/de/danoeh/antennapod/core/service/playback/PlaybackService.java

```diff
@@ -437,12 +433,7 @@ private void handleKeycode(int keycode, int source) {
                 break;
             case KeyEvent.KEYCODE_MEDIA_PAUSE:
                 if (status == PlayerStatus.PLAYING) {
-                    mediaPlayer.pause(false, true);
-                }
-                if (UserPreferences.isPersistNotify()) {
-                    mediaPlayer.pause(false, true);
-                } else {
-                    mediaPlayer.pause(true, true);
+                    mediaPlayer.pause(() -> !UserPreferences.isPersistNotify(), false);
```

Comment:
I never quite understood why we had so many calls to mediaPlayer.pause(). My only plausible guess was that we wanted to pause the playback as soon as possible, and only then check for user preferences.
Also, I fail to see why we want to reinitialize the mediaplayer pretty much every time we're pausing. This behavior doesn't really make sense when casting, but if you see it differently, then we can pass a different argument when casting and when not.